File: synchro-mes/backend/app/schemas/oee.py
"""
Schemas: OEE
"""
from pydantic import BaseModel
from datetime import date, datetime


class OeeHistoryRead(BaseModel):
    id: int
    machine_code: str
    date: date
    shift: str | None = None
    availability: float
    performance: float
    quality_rate: float
    oee: float
    planned_time_minutes: float
    running_time_minutes: float
    downtime_minutes: float
    total_produced: int
    good_produced: int
    rejected: int

    model_config = {"from_attributes": True}


class OeeSummary(BaseModel):
    """Resumo OEE para cards do dashboard."""
    machine_code: str
    oee: float
    availability: float
    performance: float
    quality_rate: float
    trend: str = "stable"  # up, down, stable
    period: str = "today"

File: frontend/src/types/oee.ts
// ── OEE ──────────────────────────────────────────────────────

export type OeeTrend = 'up' | 'down' | 'stable';

export interface OeeHistory {
  id: number;
  machine_code: string;
  date: string;
  shift: string | null;
  availability: number;
  performance: number;
  quality_rate: number;
  oee: number;
  planned_time_minutes: number;
  running_time_minutes: number;
  downtime_minutes: number;
  total_produced: number;
  good_produced: number;
  rejected: number;
}

export interface OeeSummary {
  machine_code: string;
  oee: number;
  availability: number;
  performance: number;
  quality_rate: number;
  trend: OeeTrend;
  period: string;
}
